packages/salesforcedx-apex-replay-debugger/src/core/logContext.ts

```diff
@@ -303,6 +351,9 @@ export class LogContext {
   }
 
   public updateFrames(): void {
+    if (this.hasHeapDump()) {
```

Comment:
`updateFrames` is where we continue parsing the debug log when the user steps or runs the debugger. So we revert back to the saved state.

Reply:
How big is the save state? I assume we do a rather deep copy to copy all the values?

Reply:
Yea, I have to do a deep copy so the size of the save state depends on how much variables they've built up from the debug log :grimacing: . It probably isn't as bad as it sounds because variables from the debug log are limited (e.g. concatenated strings, collections are just the toString representation).